### DISCUSSION TITLE
Swift-Perception fails to compile using Xcode 16.1 beta 2

### DISCUSSION THREAD

**Discussion by easiwriter** [pointfreeco/swift-composable-architecture#3444] [General]
Getting this error:

Failed to build module 'Charts'; this SDK is not supported by the compiler (the SDK is built with 'Apple Swift version 6.0 effective-5.10 (swiftlang-6.0.0.7.41 clang-1600.0.24.1)', while this compiler is 'Apple Swift version 6.0 effective-5.10 (swiftlang-6.0.0.9.11 clang-1600.0.26.2)'). Please select a toolchain which matches the SDK.

**Comment by mbrandonw** [comment]
Hi @easiwriter, this is a bug in beta 2 that was fixed in beta 3. If you update Xcode it should be fixed

**Reply by easiwriter** [reply]
Thanks. I just tried beta 3 and this fails too, but in a different way. 

![Screenshot 2024-10-16 at 13 15 45](https://github.com/user-attachments/assets/0ec33bb1-4a2c-4517-ae3f-acf4f3d09725)

**Reply by mbrandonw** [reply]
Please make sure you are on the newest release of swift-navigation: [2.2.2](https://github.com/pointfreeco/swift-navigation/releases/tag/2.2.2). I hope you can understand that if you are using beta software there are going to be more potential problems. If you would like to avoid these problems in the future, you can stick to the official Xcode releases.

**Comment by easiwriter** [comment]
Thanks, I normally use Xcode 16.0 but was trying the betas to see if they helped clear up my modularising problems. I usually stay clear of the betas because of potential problems, not least that youcan't publish anything to the appStore.
